src/main/java/com/ericsson/ei/queryservice/ProcessAggregatedObject.java

```diff
@@ -57,23 +57,50 @@
     public ArrayList<String> processQueryAggregatedObject(String id) {
         ObjectMapper mapper = new ObjectMapper();
         String condition = "{\"_id\" : \"" + id + "\"}";
-        log.info("The condition is : " + condition);
+        LOGGER.debug("The condition is : " + condition);
         JsonNode jsonCondition = null;
         try {
             jsonCondition = mapper.readTree(condition);
         } catch (Exception e) {
-            log.error(e.getMessage(), e);
+            LOGGER.error(e.getMessage(), e);
         }
-        log.info("The Json condition is : " + jsonCondition);
+        LOGGER.debug("The Json condition is : " + jsonCondition);
         ArrayList<String> response = handler.find(aggregationDataBaseName, aggregationCollectionName,
                 jsonCondition.toString());
         return response;
     }
+    
+    /**
+     * The method is responsible to extract the aggregated data on the basis of
+     * the ID from the aggregatedObject.
+     * 
+     * @param id
+     * @return ArrayList
+     */
+    public ArrayList<String> getAggregatedObjectByTemplateName(String templateName) {
+        String condition = "{\"_id\": /.*" + templateName + "/}";
+        LOGGER.debug("The Json condition is : " + condition);
+        ArrayList<String> response = handler.find(aggregationDataBaseName, aggregationCollectionName, condition);
+        return response;
```

Comment:
Same here. Or are we doing this extra step to show more clearly what type that is returned?